OrbitQt/AccessibilityAdapter.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "AccessibilityAdapter.h"
+
+#include <QWidget>
+
+namespace orbit_qt {
+
+absl::flat_hash_map<const orbit_gl::GlAccessibleInterface*, QAccessibleInterface*>
+    A11yAdapter::interface_map_;
+bool A11yAdapter::initialized_ = false;
+
+//================ A11yAdapter ==================
+
+void A11yAdapter::Init() {
+  orbit_gl::GlAccessibleInterfaceRegistry::Get().OnUnregistered(
+      [](orbit_gl::GlAccessibleInterface* iface) {
+        if (interface_map_.contains(iface)) {
+          auto adapter = dynamic_cast<A11yAdapter*>(interface_map_[iface]);
```

Comment:
Yes - this also covers your comment below. I'm not checking if this element exists, I'm checking if this element can be cast to a A11yAdapter*. `dynamic_cast` will return a nullptr if the types don't match.

This is done to track elements that need to be destroyed when they are unregistered: A11yAdapter creates new adapters on `GetOrCreateAdapter`, and destroys them here. But there may be objects in that map that are not created by A11yAdapter (namely the OrbitGlWidgetAccessible, which is created by the Qt System) that should not be destroyed.

I could add a comment to make this clear.

Reply:
I'm not a huge fan of RTTI. The style guide discourages it also: https://google.github.io/styleguide/cppguide.html#Run-Time_Type_Information__RTTI_. I would favor an alternative if possible.

Reply:
Oh, I wasn't aware that our style guide is discouraging RTTI. It's quite straightforward to replace in this case, I will add a static set to `AccessibilityAdapter` that will explicitly track resources created through `GetOrCreateAdapter`, then we don't need to rely on RTTI.

Reply:
Hm. Actually I'm not sure if this is nicer, because it also contains a bit of an ugly workaround due to the fact that the destructor of `QAccessibleInterface` is protected - PTAL at the last commit and let me know if this is better or worse than the RTTI in your opinion...

Reply:
Davide's comments below lead to a better solution for this which doesn't require the ugly workaround any more, FYI!